lib/opal/nodes/top.rb

```diff
@@ -104,9 +104,9 @@ def closing
             # require absolute paths from CLI. For other cases
             # we can expect the module names to be normalized
             # already.
-            line "Opal.load_normalized(#{module_name.inspect});"
+            line "Opal.queue(()=>{ Opal.load_normalized(#{module_name.inspect}) });"
```

Comment:
I don't really remember this code, will take a closer look at a proper review. But - async top requires need to be `require("something").__await__`.

Ie. won't this break such code?

```ruby
# file1.rb
$a = 1
require "file2"
$a == 2 or raise "Wrong result! Expected 2, got 1!"

# file2.rb
$a = 2
```

If it won't, what about:

```ruby
# file1.rb
$a = 1
require("file2").__await__
$a == 2 or raise "Wrong result! Expected 2, got 1!"

# file2.rb
# await: true
$a = 2.__await__
```

Reply:
I will test with this case

Reply:
I created file1.rb as above and added a `puts $a` after the `require "file2"` line
I created file2.rb as above
I compiled with `bundle exec opal -c -I. -o file1.js file1.rb`
I executed with `node file1.js`
It printed: `2`

So, seems to work

Reply:
First case, without __await__

Reply:
Looks like the second case also works